### ISSUE TITLE
Parent block selector button: missing focus style when 'Show button text labels' is enabled

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57710]
### Description

The Parent block selector button doesn't show a focus style when 'Show button text labels' is enabled.

Focus indication must always be clear. It appears it broke after https://github.com/WordPress/gutenberg/pull/56335

Turns out all the buttons must use `position: relative` to allow the CSS pseudo generated element responsible for the focus style to appear and be positioned correctly. https://github.com/WordPress/gutenberg/pull/56335 set `position: static` on this button, thus breaking the focus style.

Screenshot from WordPress 6.4:

![6 4](https://github.com/WordPress/gutenberg/assets/1682452/c9ade978-1c5f-4b54-8352-13b6b4fd3728)

Screenshot from Gutenberg trunk:

![trunk](https://github.com/WordPress/gutenberg/assets/1682452/e350b81e-981c-4295-968a-66579deebe5b)

Note: the button label is now longer, it changed in https://github.com/WordPress/gutenberg/pull/56146

When the 'Top toolbar' is enabled, the focus style is visible. Screenshot:

![Screenshot 2024-01-10 at 09 57 22](https://github.com/WordPress/gutenberg/assets/1682452/38199d50-e4c5-4d7e-9620-0f4512eccf6b)

## Other visual glitches

Left and right padding is inconsistent compared with other buttons. Screenshot:

![Screenshot 2024-01-10 at 10 00 55](https://github.com/WordPress/gutenberg/assets/1682452/36bb0b9a-8b76-4b45-bbb0-f71bc060dd92)

When 'Top toolbar' is enabled, a 'dot divider' visually separates the parent selector button from the other buttons. The dot is unevenly positioned, with both icons or text displayed. Screenshots:

![Screenshot 2024-01-10 at 09 18 12](https://github.com/WordPress/gutenberg/assets/1682452/fbd83044-446e-4ad0-8c79-59a80c9bd376)

![Screenshot 2024-01-10 at 09 32 44](https://github.com/WordPress/gutenberg/assets/1682452/bba9f722-cd51-408b-a6fa-e45611f2221e)

### Step-by-step reproduction instructions

- Enable: Options > Preferences > Accessibility > 'Show button text labels'
- Add a Group block and add a few paragraphs within it.
- Select one of the paragraphs.
- Shift+Tab to move focus to the block toolbar.
- Observe the parent selector button doesn't show any focus style.
- Visual glitches: see screenshots.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Also: the font size of the 'block switcher' button (the text 'Paragraph' in the screenshot below) is clearly bigger than the font size of the other buttons. It's 14 pixels vs. 12 pixels. This inconsistency doesn't help with aligning the buttons text. Seems simple enough to fix in this issue.

![Screenshot 2024-01-10 at 14 55 40](https://github.com/WordPress/gutenberg/assets/1682452/5dbf8102-fd19-4b58-a21d-0df452c89971)

The font size inconsistency is visible also in the Site Editor:

![Screenshot 2024-01-10 at 15 59 26](https://github.com/WordPress/gutenberg/assets/1682452/61b50b22-1596-40b1-9150-0090b40bbdb3)

Introduced in https://github.com/WordPress/gutenberg/pull/28691 for the template part 'display title'. I'm not sure that is still in use and anyways it shouldn't be done this way.

**Comment by afercia** [comment]
Just noticed one more problem: wiht 'Top toolbar' and 'Show icon labels' enabled, the block toolbar buttons shift vertically by 1 pixel when moving focus between the buttons 'Move Up' and 'Moce Down'. See animated Gif below.

This is triggered ny some overflow CSS properties applied to the toolbar:
```
    overflow: auto;
    overflow-y: hidden;
```

There's two problems with this:
- The vertical shift should not happen.
- More importantly: with Firefox there is a silent tab stop between the document tools toolbar and the block toolbar. Firefox makes focusable all scrollable elements that have an actual content overflow. As such, the div element with overflow is focusable in Firefox. This is far from ideal for keyboard accessibility and needs to be fixed. Will create a separate issue.

Animated GIF to illustrate the vertical shift on current trunk:

![shift](https://github.com/WordPress/gutenberg/assets/1682452/309c61fd-12a9-4223-aa53-9b1b08706c02)
